flaml/automl.py

```diff
@@ -1026,12 +1026,27 @@ def _validate_ts_data(
             assert (
                 dataframe[[dataframe.columns[0]]].duplicated() is None
             ), "Duplicate timestamp values with different values for other columns."
-        ts_series = pd.to_datetime(dataframe[dataframe.columns[0]])
-        inferred_freq = pd.infer_freq(ts_series)
-        if inferred_freq is None:
-            logger.warning(
-                "Missing timestamps detected. To avoid error with estimators, set estimator list to ['prophet']. "
-            )
+        if self._state.task == TS_FORECASTPANEL:
+            # check for each time series independently
+            group_ids = self._state.fit_kwargs.get("group_ids")
+            unique_ids = dataframe[group_ids].value_counts().reset_index()[group_ids]
+            for _, row in unique_ids.iterrows():
+                df = dataframe.copy()
+                for id in group_ids:
+                    ts = df.loc[df[id] == row[id]]
+            ts_series = pd.to_datetime(ts[ts.columns[0]])
+            inferred_freq = pd.infer_freq(ts_series)
+            if inferred_freq is None:
+                logger.warning(
+                    "Missing timestamps detected. To avoid error with estimators, set estimator list to ['prophet']. "
+                )
```

Comment:
There are multiple problems:
1. Does prophet handle panel data? Does TFT handle missing timestamps? If TFT can handle missing timestamps, this check is not necessary. If prophet can't handle panel data, this message shouldn't suggest adding prophet.
2. For loop is not efficient. There should be a functional way using `groupby()`.
3. Do you intend to infer the frequency for each series? The current code only does it for the last one.

Reply:
To clarify, TimeSeriesDataset only handle timestamps that are missing, but do not handle `NA` values.

Reply:
@EgorKraevTransferwise and @markharley Our current solution to this issue will be to either require no missing timestamps or else user have to provide a `freq` argument.